src/app/plan/duck/sagas.ts

```diff
@@ -287,14 +287,20 @@ function* checkUpdatedPVs(action) {
 
       if (updatedPlan) {
         const isUpdatedPVList = () => {
-          const updatedGeneration = updatedPlan.metadata.generation;
-          const oldGeneration = currentPlan.metadata.generation;
-
-          //Generation check incremented twice: once for ui change, once for controller change.
-          if (isRerunPVDiscovery) {
-            return updatedGeneration >= oldGeneration + 2;
+          if (updatedPlan.status) {
+            if (isRerunPVDiscovery) {
+              const updatedGeneration = updatedPlan.status.observedDigest;
+              const oldGeneration = currentPlan.status.observedDigest;
+              if (updatedGeneration === oldGeneration) {
```

Comment:
Can we update each of these return paths with a comment that describes the specific case?